resolve/index_test.go

```diff
@@ -0,0 +1,152 @@
+package resolve_test
+
+import (
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/bazel-gazelle/config"
+	"github.com/bazelbuild/bazel-gazelle/label"
+	"github.com/bazelbuild/bazel-gazelle/repo"
+	"github.com/bazelbuild/bazel-gazelle/resolve"
+	"github.com/bazelbuild/bazel-gazelle/rule"
+)
+
+// testResolver implements the resolve.Resolve interface, with a configurable language name and
+// Imports/Embeds taken from the rule attributes.
+type testResolver struct{ name string }
+
+func (tr testResolver) Name() string { return tr.name }
+
+// imports is expected to be a list of the form ["Lang|Imp"]
+func (tr testResolver) Imports(c *config.Config, r *rule.Rule, f *rule.File) []resolve.ImportSpec {
+	if r.Attr("imports") == nil {
```

Comment:
This check is here to force Imports() to return nil in that case, rather than an empty slice. The semantics of Imports makes those two things different, and so we mimic that here by having the semantics of `imports=[]` not be the same as if `imports` is missing entirely.